httpsupport/http_client_blackbox_test.go

```diff
@@ -0,0 +1,94 @@
+package httpsupport_test
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-common/httpsupport"
+	"github.com/fabric8-services/fabric8-common/resource"
+
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHost(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	req := &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+
+	t.Run("prod mode", func(t *testing.T) {
+		host := httpsupport.Host(req, &dummyConfig{})
+		assert.Equal(t, "api.service.domain.org", host)
+	})
+
+	t.Run("nil config", func(t *testing.T) {
+		host := httpsupport.Host(req, nil)
+		assert.Equal(t, "api.service.domain.org", host)
+	})
+
+	t.Run("dev mode", func(t *testing.T) {
+		host := httpsupport.Host(req, &dummyConfig{true})
+		assert.Equal(t, "auth.openshift.io", host)
+	})
+}
+
+func TestAddParams(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	t.Run("success", func(t *testing.T) {
+		testMap := map[string]string{
+			"param1": "a",
+			"param2": "b",
+			"param3": "https://www.redhat.com",
+		}
+		testHost := "openshift.io"
+		generatedURLString, err := httpsupport.AddParams("https://"+testHost, testMap)
+		require.NoError(t, err)
+
+		generateURL, err := url.Parse(generatedURLString)
+		require.NoError(t, err)
+
+		assert.Equal(t, testHost, generateURL.Host)
+		assert.Equal(t, "https", generateURL.Scheme)
+
+		m, _ := url.ParseQuery(generateURL.RawQuery)
+		for k, v := range testMap {
+			assert.Equal(t, v, m[k][0])
+		}
```

Comment:
why you are processing `generatedURLString` and then asserting different thing.  I guess, better to assert with `strings.Contains()` and `strings.HasPrefix` etc ??